File: app/src/main/java/com/aura/connection/BankService.java
package com.aura.connection;

import okhttp3.Credentials;
import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface BankService {


    @POST ("login/{id}{password}")
    Call<Credentials> getLogin(@Path("id,password") String id, String password);

    public static final Retrofit retrofit = new  Retrofit.Builder()
            .baseUrl("http://192.168.1.84")
            .addConverterFactory(GsonConverterFactory.create())
            .build();


}

File: app/src/main/java/com/aura/connection/BankService.kt
package com.aura.connection


import com.aura.model.CredentialsResult
import com.aura.model.Account
import com.aura.model.Credentials
import com.aura.model.Transfer
import com.aura.model.TransferResult
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface BankService {
    @POST("/login")
    suspend fun getLogin(
        @Body credentials: Credentials
    ): Response<CredentialsResult>


    @GET("/accounts/{id}")

    suspend fun getAccount (
        @Path("id") accountId: String

    ): Response<List<Account>>


    @POST("/transfer")
    suspend fun getTransfer(
        @Body transferRequest: Transfer
    ): Response<TransferResult>



    companion object {

        private const val api_base_url: String = "http://10.0.2.2:8080"

        val retrofit: Retrofit = Retrofit.Builder()
            .baseUrl(api_base_url)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }
}






